store/store_test.go

```diff
@@ -613,12 +614,63 @@ func TestStoreInit(t *testing.T) {
 	store, err := NewStore[*headertest.DummyHeader](ds)
 	require.NoError(t, err)
 
+	err = store.Start(ctx)
+	require.NoError(t, err)
+
 	headers := suite.GenDummyHeaders(10)
 	h := headers[len(headers)-1]
-	err = store.Init(ctx, h) // init should work with any height, not only 1
+	err = store.Append(ctx, h) // init should work with any height, not only 1
 	require.NoError(t, err)
+	time.Sleep(100 * time.Millisecond)
 
 	tail, err := store.Tail(ctx)
+	require.NoError(t, err)
 	assert.Equal(t, tail.Hash(), h.Hash())
+}
+
+func TestInitStore_NoReinit(t *testing.T) {
```

Comment:
```suggestion
func TestStore_Reinit(t *testing.T) {
```

Also, TestStoreInit seems to repeat this one, so I propose to merge these two under `TestStore...smth`

Reply:
Merged into one.